services-geojson/src/main/java/com/mapbox/geojson/Feature.java

```diff
@@ -178,7 +178,7 @@ public static Feature fromGeometry(@Nullable Geometry geometry, @Nullable JsonOb
      * @return {@link Feature}
      * @since 1.0.0
      */
-    public static Feature fromGeometry(@Nullable Geometry geometry, @NonNull JsonObject properties,
+    public static Feature fromGeometry(@Nullable Geometry geometry, @Nullable JsonObject properties,
```

Comment:
Interesting. Does this mean the `properties` object is `Nullable` or a hard `NonNull`? I am assuming that `@NonNull` will enforce a if-else null check which `@Nullable` won't.

Reply:
The property itself is nullable. But there is also a getter which I made non nullable. 

The code was using the getter and it would crash anyway when it was null. I made the crash an explicit exception.

So there is an inner contract (as there was before) that the getter will only be called when the property is already set.

Reply:
So the properties parameter was marked as @NonNull while this is the place were we are ensuring that a non null properties parameter is passed to the Feature by creating a new JsonObject() when the passed in value (marked NotNullabl!) is not null. That did not make any sense to me.